asterius/src/Asterius/Types.hs

```diff
@@ -126,6 +128,18 @@ instance Semigroup AsteriusModule where
 instance Monoid AsteriusModule where
   mempty = AsteriusModule mempty mempty mempty mempty mempty
 
+-- | Given the worker thread pool capacity @c@, @parForceAsteriusModule c m@
+-- deeply evaluates an 'AsteriusModule' m in parallel on the global thread
+-- pool.
+parForceAsteriusModule :: Int -> AsteriusModule -> AsteriusModule
+parForceAsteriusModule n m@(AsteriusModule sm se fm spt mod_ffi_state) =
+  parallelRnf n (SM.toList sm)
+    `seq` parallelRnf n (SM.toList se)
+    `seq` parallelRnf n (SM.toList fm)
+    `seq` parallelRnf n (SM.toList spt)
+    `seq` rnf mod_ffi_state -- TODO: is it worth it parallelizing deeper?
```

Comment:
I think not; `mod_ffi_state` should only count for a small part of the whole `AsteriusModule`.